src/workerd/api/node/tests/http-client-nodejs-test.js

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import http from 'node:http';
+import https from 'node:https';
+import { strictEqual, ok, deepStrictEqual, throws } from 'node:assert';
+import { mock } from 'node:test';
+
+export const checkPortsSetCorrectly = {
+  test(_ctrl, env) {
+    const keys = [
+      'PONG_SERVER_PORT',
+      'ASD_SERVER_PORT',
+      'TIMEOUT_SERVER_PORT',
+      'HELLO_WORLD_SERVER_PORT',
+      'HEADER_VALIDATION_SERVER_PORT',
+    ];
+    for (const key of keys) {
+      strictEqual(typeof env[key], 'string');
+      ok(env[key].length > 0);
+    }
+  },
+};
+
+// Test is taken from Node.js: test/parallel/test-http-client-request-options.js
+export const testHttpClientRequestOptions = {
+  async test(_ctrl, env) {
+    const headers = { foo: 'Bar' };
+    const { promise, resolve } = Promise.withResolvers();
+    const url = new URL(`http://localhost:${env.PONG_SERVER_PORT}/ping?q=term`);
+    url.headers = headers;
+    const clientReq = http.request(url);
+    clientReq.on('close', resolve);
+    clientReq.end();
+    await promise;
+  },
+};
+
+// Test is taken from test/parallel/test-http-client-res-destroyed.js
+export const testHttpClientResDestroyed = {
+  async test(_ctrl, env) {
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      http.get(
+        {
+          port: env.ASD_SERVER_PORT,
+        },
+        (res) => {
+          strictEqual(res.destroyed, false);
+          res.destroy();
+          strictEqual(res.destroyed, true);
+          res.on('close', resolve);
+        }
+      );
+      await promise;
```

Comment:
Just noting, it's going to be critical that we document this pattern on adding a promise on the close event that is awaiting in the fetch handler, either directly like this or via the `ctx.waitUntil(...)`. Keep in mind, however, that the `'error'` event is emitted after the `'close'` event, so it's at least theoretically possible that resolving on `close` could cause the error to not be handled. We should test that case if there's not a test in here already.